components/org.wso2.carbon.deployment.notifier/src/main/java/org/wso2/carbon/deployment/notifier/DeploymentNotifierLifecycleListener.java

```diff
@@ -59,8 +61,15 @@
     String serverId;
 
     public DeploymentNotifierLifecycleListener() {
-        config = DeploymentConfigurationProvider.
-                getDeploymentConfiguration().getDeploymentNotifier();
+        try {
+            ConfigProvider configProvider = DataHolder.getInstance().getConfigProvider();
```

Comment:
Instead of getting the ConfigProvider through the DataHolder, we can make this class a declarative service by adding @Component class-level annotation. Then, we can wait for the ConfigProvider OSGi Service via @Reference annotation.